internal/metrics/export/otel.go

```diff
@@ -29,7 +29,7 @@ const (
 )
 
 type OTelExporter struct {
-	conf           config.Config
+	conf           *config.Config
```

Comment:
Are we sure that we wanna pass a pointer to the config around instead of copying it around? I'd think a config struct should be immutable after it has been read from file.

Reply:
It was inconsistent through the project. Being immutable is great, but I don't think we should have additional resources through copies of structs and it's more difficult if things change to update all the copies when in reality the config should be read once and only modified in some places